File: android/src/main/kotlin/com/giphy/giphy_flutter_sdk/GiphyFlutterGridViewFactory.kt
package com.giphy.giphy_flutter_sdk

import io.flutter.plugin.platform.PlatformView
import io.flutter.plugin.platform.PlatformViewFactory
import io.flutter.plugin.common.StandardMessageCodec
import android.content.Context
import io.flutter.plugin.common.BinaryMessenger

class GiphyFlutterGridViewFactory(private val messenger: BinaryMessenger) :
    PlatformViewFactory(StandardMessageCodec.INSTANCE) {
    override fun create(context: Context?, viewId: Int, args: Any?): PlatformView {
        val creationParams = args as Map<String, Any>?
        return GiphyFlutterGridView(context, messenger, viewId, creationParams)
    }
}

File: ios/giphy_flutter_sdk/Sources/giphy_flutter_sdk/GiphyFlutterGridViewFactory.swift
//
//  GiphyFlutterGridViewFactory.swift
//  giphy_flutter_sdk
//
//  Created by Alex on 4.3.2024.
//

import Flutter
import UIKit

public class GiphyFlutterGridViewFactory: NSObject, FlutterPlatformViewFactory {
    private var messenger: FlutterBinaryMessenger
    
    public init(messenger: FlutterBinaryMessenger) {
        self.messenger = messenger
        super.init()
    }
    
    public func create(withFrame frame: CGRect, viewIdentifier viewId: Int64, arguments args: Any?) -> FlutterPlatformView {
        return GiphyFlutterGridView(frame: frame, viewId: viewId, args: args, binaryMessenger: messenger)
    }
}
